power_grid_model_c/power_grid_model/include/power_grid_model/prepare_calculate.hpp

```diff
@@ -0,0 +1,119 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+
+#include "topology.hpp"
+
+#include "common/common.hpp"
+
+#include "math_solver/math_solver_dispatch.hpp"
+
+#include "main_core/main_model_type.hpp"
+#include "main_core/math_state.hpp"
+#include "main_core/topology.hpp"
+#include "main_core/y_bus.hpp"
+
+namespace power_grid_model {
+struct SolverPreparationContext {
+    main_core::MathState math_state;
+    MathSolverDispatcher const* math_solver_dispatcher;
+};
+
+template <class ModelType>
+    requires(main_core::is_main_model_type_v<ModelType>)
+struct StatusCheckingContext {
+    bool is_topology_up_to_date{false};
+    bool last_updated_calculation_symmetry_mode{false};
+    typename ModelType::SequenceIdx parameter_changed_components{};
```

Comment:
it feels a little bit confusing to me that the status and the cache are part of the same struct called `StatusCheckingContext`.

Maybe the name is just not clear enough. Maybe something along the lines of `CacheStatus` or `CacheValidity`?

That said, maybe the confusing part is not the name of the type but the fact that one thing tells something about the cache validity while the other about the context in which it is valid? maybe that means that they shouldn't be part of the same class in the first place?